docs/building-spark.md

```diff
@@ -9,6 +9,10 @@ redirect_from: "building-with-maven.html"
 
 Building Spark using Maven requires Maven 3.0.4 or newer and Java 6+.
 
+**Note:** Building Spark with Java 7 or later can create JAR files that may not be
```

Comment:
The "or later" part is not true, I think. I remember testing that JDK7 is the problem: jars with > 65k files created on JDK7 won't work on either JDK6 *or* JDK8 (!). Jars created with anything but JDK7 seem to work everywhere.